### DISCUSSION TITLE
Is there a way to enable columnFilterModeOptions when I have a custom Filter component?

### DISCUSSION THREAD

**Discussion by ashatat** [KevinVandy/material-react-table#475] [Q&A]
```js
const columns = [
  {
    header: "birth date",
    accessorKey: "bday",
    columnFilterModeOptions: ["between", "lessThan", "greaterThan"],
    Filter: ({ column }) => {
	    return (
		    <DateRangePicker
			    onChange={(newValue) => {
				    column.setFilterValue(newValue);
			    }}
			    value={column.getFilterValue()}
		    />
	    );
    },
  }
];
```

**Comment by Rasool-Abbas** [comment]
I am in need of an answer to this.

**Reply by KevinVandy** [reply]
You'll just have to render the menu yourself in an input adornment.

**Reply by Rasool-Abbas** [reply]
Understood. That was resolved, thank you. 

After some more time looking into it, i couldn't find any documentation regarding how you would change the table's columnFilterModeOptions from the custom filter component. I found by trial and error that we needed to set the tables state directly from the table reference that Filter exposes. I had some trouble finding this, so I will leave it here if anyone is stuck on a similar issue.
